### PR TITLE
Add Metadata and LinkedData to IDX pages

### LINKED ISSUE #6415: Add Metadata to IDX pagetype for persian/afghanistan
**Is your feature request related to a problem? Please describe.**
IDX pages should contain the correct metadata

**Given**: I am on the url /persian/afghanistan
**When**: I share the Page on Twitter and Facebook
**Then**: I get the correct Title and Description

**Describe the solution you'd like**
Use `summary` tag from ares as the `description` meta tag
The `title` meta tag should be the same as the title displayed in the h1 in https://github.com/bbc/simorgh/issues/6414.
Otherwise display the same information as the homepage. We should then audit this.

**Describe alternatives you've considered**

**Testing notes**
This will need unit tests.
Create metadata tests following an example (these are covered by the core integration tests that will be run for this page type - these need to be added in another issue.)

- [x] This feature is expected to need manual testing.

**Additional context**

### LINKED ISSUE #6621: Add Linked data to IDX page
**Is your feature request related to a problem? Please describe.**
We currently have an IDX skeleton page and need to add Linked Data to this page.

**Describe the solution you'd like**
- Import the linked data container into the IDX page.

The linkedData currently has [props](https://github.com/bbc/simorgh/blob/b11344b1461700bdfd3734a7c1a5c2bee786f2b2/src/app/containers/LinkedData/index.jsx#L99-L115) of: 
```
LinkedData.propTypes = {
  showAuthor: bool,
  type: string.isRequired,
  seoTitle: string.isRequired,
  headline: string,
  description: string,
  datePublished: string,
  dateModified: string,
  aboutTags: arrayOf(
    shape({
      '@type': string,
      name: string,
      sameAs: arrayOf(string),
    }),
  ),
};
```

We should pass the same props as what the FrontPage does (`type` and `seoTitle`). The seoTitle should be the meta title.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by itimirichard**
Resolves #6415 and https://github.com/bbc/simorgh/issues/6621

**Overall change:**
_Add metadata and linked-data to IDX pages._

**Code changes:**

- _Import and use MetadataContainer and LinkedData._
- _pass `lang, title and description` props to MetadataContainer from `pageData.metadata`._
- _pass `seoTitle` props to LinkedData from `pageData.promo`._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

**lighthouse test:**
- Currently on test 
Screenshot 2020-06-03 at 9 40 54 AM

- on this PR
Screenshot 2020-06-03 at 9 41 33 AM

**Structured data (linked data) test:**
Screenshot 2020-06-03 at 9 45 34 AM

**AMP Validator test:**
Screenshot 2020-06-03 at 9 48 10 AM

### CODE REVIEW

File: src/app/pages/IdxPage/index.jsx [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
-        ))}
-      </PageContainer>
-    </main>
+    <>
+      <MetadataContainer
+        title={title}
+        lang={lang}
+        description={summary}
+        openGraphType="website"
+      />
```
Comment (sareh) [inline src/app/pages/IdxPage/index.jsx:61]: The metadata also includes LinkedData. We should ensure we have the same linked data for these IDX pages as for the front page. 
```
const seoTitle = path(['promo', 'name'], pageData);
```


```suggestion
      />
      <LinkedData type="WebPage" seoTitle={seoTitle} />
```

Comment (itimirichard) [reply src/app/pages/IdxPage/index.jsx:61]: Thanks @sareh, for pointing that out.

**Comment by itimirichard** [comment]
I'm wondering if this would need manual testing. How do we typical test validate the metadata and linkeddata?

File: src/app/pages/IdxPage/index.jsx (hunk 1 above)
Comment (DenisHdz) [reply src/app/pages/IdxPage/index.jsx:61]: There was a separate issue for that https://github.com/bbc/simorgh/issues/6621. Can you link it to this PR and let Ruth know, please?

Comment (itimirichard) [reply src/app/pages/IdxPage/index.jsx:61]: Oh, I had no idea there was an issue for that, thank you Denis, I'll inform Ruth.

**Comment by sareh** [comment]
@itimirichard We usually use tools to check metadata:

You can run the **Lighthouse** (in Chrome dev tools) manually with this branch locally on your machine - it should have no metadata related errors

You can run the **AMP validator** manually with this branch locally on your machine:
`npx amphtml-validator http://localhost:7080/persian/afghanistan.amp` & 
`npx amphtml-validator http://localhost:7080/ukrainian/ukraine_in_russian.amp`

Screenshotting the output and putting it on this PR would be good. 

Just so you're aware, AMP validator can be run on each build via `npm run amp:validate` but this uses urls that are in the cypress config: `cypress/support/config/settings.js`.

When changes are on Test/Live they can be checked via this url https://validator.ampproject.org/#url=https%3A%2F%2Fwww.test.bbc.com%2Fpersian%2Fafghanistan.amp


You chan check **Structured Data** (linked data) here: https://search.google.com/structured-data/testing-tool/u/0/ 
You can paste in the markup as a 'code snippet' and run the tool. When the LinkedData component is added, you'll be able to see information about the 'WebPage' type in the output.

**Review by EinsteinNjoroge — COMMENTED** [review]
@itimirichard I would expect this to change some snapshots, Is that not the case?

**Comment by itimirichard** [comment]
> @itimirichard I would expect this to change some snapshots, Is that not the case?

@EinsteinNjoroge No that isn't the case, this didn't change any snapshots. I presume that this is because these are added as meta tags to the `head` element through helmet

**Comment by sareh** [comment]
@itimirichard @EinsteinNjoroge 
We have integration tests with snapshots for metadata. e.g. https://github.com/bbc/simorgh/blob/latest/src/integration/common/SEO.js
I would expect that we should ensure we're running our integration tests here for these IDX pages https://github.com/bbc/simorgh/tree/latest/src/integration/pages

cc @PriyaKR

**Review by DenisHdz — APPROVED** [review]
Looks good! 

Can you also add a screenshot with the Lighthouse scores for Accessibility? We should expect it to be higher than 83 since this also adds <title> to the page.

Screenshot 2020-06-02 at 15 52 57

**Review by thekp — APPROVED** [review]
LGTM!

**Comment by itimirichard** [comment]
> Looks good!
> 
> Can you also add a screenshot with the Lighthouse scores for Accessibility? We should expect it to be higher than 83 since this also adds <title> to the page.
> 
> Screenshot 2020-06-02 at 15 52 57

Yes, its definitely higher now. Screenshot below

Screenshot 2020-06-03 at 2 14 49 PM

**Comment by PriyaKR** [comment]
Looks good to me across browsers and devices as this PR adds the viewport details.

Noticed one thing in opera mini extreme mode but this issue is across frontpage too.Not something caused by this PR.
Opera_mini_extreme_mode_frontpage
Opera_mini_extreme_mode

**Pull request merged**